### DISCUSSION TITLE
Texlab snippets not shown

### DISCUSSION THREAD

**Discussion by hnorkowski** [helix-editor/helix#9139] [General] [answered]
### Summary

When editing a tex-file using hx with texlab LSP there are no snippets shown from the LSP.
However, they work with the same LSP (and version) in neovim and vscode.

### Reproduction Steps

 

I tried this:

- Opening tex file in `hx` with texlab LSP and typing `\begin` on a new line

I expected this to happen:

- Autocomplete offering a begin snippet that also adds `\end` and let me edit the name of the environment at both places at once

Instead, this happened:

- No snippet shown
- While testing it randomly worked a single time but never again.

Recoding: https://asciinema.org/a/kSEMnunY6qDjA2yEB80AkJtXm

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>
Log exceeds 65536 chars:
https://pastebin.com/ZGEbZTRR
</details>

### Platform

Linux

### Terminal Emulator

alacritty 0.12.3 (5efb0690)

### Installation Method

Arch AUR: helix-git

### Helix Version

helix 23.10 (d5e6749f)

**Comment by pascalkuthe** [comment, marked as answer]
Texlab uses incomplete completion rerequests (and isn't nearly as good at guessing what you want as other severs like RA). Helix, doesn't support these yet, that is a known limitation. You can press c-x to request new completions which will usually contain what you need.

Once #8021 is merged we can implement that part of the spec so the request is automatic

**Reply by yutanagano** [reply]
Hi- I had the same issue and initially the marked answer here did work- but somehow control-x is no longer showing completions in helix with texlab. I can't tell whether this is because of an update in helix or in texlab, but using newest versions (helix v24.3 and texlab v5.15) this no longer works for me.

**Comment by lukeflo** [comment]
Since I had a similiar problem with `texlab`, here another solution: 

A Latex LSP with working built-in snippets is [digestif](https://github.com/astoff/digestif?tab=readme-ov-file).

After installing it following the instructions on the GitHub page, its as simple as putting the following lines into your `languages.toml`:

```toml
#digestif
[language-server.digestif]
command = "digestif"

[[language]]
name = "latex"
# language-servers = [ "texlab", "scls" ] # old texlab config
language-servers = [ "digestif", "scls" ]
```

If you now open a `.tex` document and type `\beg...` it will suggest to autocomplete an environment. It then inserts the `begin` and `end` part and lets you fill out the env-name with two cursors, here typical names as `itemize` etc. will again be autocompleted.

Plus, if your `INFOPATH` contains the location of your current TeXLive doc files, you can open a popup with detailed informations describing the macro under the cursor with <kbd>space</kbd>+<kbd>k</kbd>.

Autocompletion for literature references etc. is also included.

For me it works way better than `texlab`; at least, for the moment...
